x-pack/filebeat/input/http_endpoint/input_test.go

```diff
@@ -288,6 +343,18 @@ func TestServerPool(t *testing.T) {
 	}
 }
 
+func doRequest(method, url, contentType string, body io.Reader) (*http.Response, error) {
+	if method == "" {
+		method = http.MethodPost
+	}
+	req, err := http.NewRequest(method, url, body)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", contentType)
```

Comment:
Do we need to check `Content-Length > 0` to make sure we always `request.body`?

Reply:
This doesn't change the behaviour at all except for allowing PUT and PATCH requests to be used, so I don't think so.